tbdex/offering/offering.go

```diff
@@ -47,9 +47,35 @@ type PayoutDetails struct {
 	Methods      []PayoutMethod `json:"methods,omitempty"`
 }
 
+// PaymentMethod is an interface for PayinMethod and PayoutMethod.
```

Comment:
i declared an interface for both PayinMethod and PayoutMethod to snap to, just so that i can write this:

```go
func (r *RFQ) verifyPaymentMethod(
	selectedPaymentKind string,
	selectedPaymentDetailsHash string,
	privateData *PrivateData,
	offeringPaymentMethods []offering.PaymentMethod, // <- this part
	payDirection PayDirection,
) error
```

without this, i would have had to resort to writing this function twice, one for verifyPayinMethod that passes in `offeringPayinMethods []offering.PayinMethod` and verifyPayoutMethod that passes in `offeringPayoutMethods []offering.PayoutMethod` 

go does not support type inheritance and this was my best attempt at it. open to other suggestions tho